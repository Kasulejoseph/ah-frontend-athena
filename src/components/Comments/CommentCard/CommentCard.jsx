import React from "react";
import PropTypes from "prop-types";
import { MDBInput, MDBContainer, MDBCardBody, MDBCard, MDBBtn } from "mdbreact";

const CommentCard = ({ onChange, onSubmitReply }) => {
  return (
    <div className="">
      <MDBContainer className="my-3 mx-0 px-0">
        <MDBCard className="w-75">
          <MDBCardBody>
            <form onSubmit={onSubmitReply}>
              <MDBInput
                onChange={onChange}
                label="add comment"
                name="comment_body"
                type="textarea"
                id="reply"
                cols="30"
                rows="2"
                required
              />
              <p />
              <MDBBtn
                className="m-1 p-1 left ml-4"
                color="primary"
                type="submit"
              >
                comment
              </MDBBtn>
            </form>
          </MDBCardBody>
        </MDBCard>
      </MDBContainer>
    </div>
  );
};

CommentCard.propTypes = {
  onChange: PropTypes.func,
  onSubmitReply: PropTypes.func
};
CommentCard.defaultProps = {
  onChange: () => {},
  onSubmitReply: () => {}
};
export default CommentCard;
