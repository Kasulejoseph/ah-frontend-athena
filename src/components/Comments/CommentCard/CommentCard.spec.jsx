import { shallow} from "enzyme";
import React from "react";
import CommentCard from "./CommentCard";

const props = {
  onChange: jest.fn(),
  onSubmitReply: jest.fn()
};
const mockCommentCard = jest.fn();
let wrapper;


describe("CommentCard test", () => {
  describe("CommentCard test", () => {
    beforeEach(() => {
      wrapper = shallow(<CommentCard {...props} />);
    });

  it("matches snapshot", () => {
    const component = shallow(<CommentCard {...mockCommentCard} />);
    expect(component).toMatchSnapshot();
  });
  it("should render a form", () => {
    expect(wrapper.find("form")).toHaveLength(1);
  });
  })
})
