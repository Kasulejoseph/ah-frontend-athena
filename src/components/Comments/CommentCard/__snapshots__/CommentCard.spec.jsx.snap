// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard test CommentCard test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentCard
    _isMockFunction={true}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    onChange={[Function]}
    onSubmitReply={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        className="my-3 mx-0 px-0"
        fluid={false}
        tag="div"
      >
        <t
          className="w-75"
          tag="div"
        >
          <t
            tag="div"
          >
            <form
              onSubmit={[Function]}
            >
              <t
                className=""
                cols="30"
                containerClass=""
                disabled={false}
                error=""
                filled={false}
                gap={false}
                group={false}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                id="reply"
                label="add comment"
                labelClass=""
                name="comment_body"
                onChange={[Function]}
                required={true}
                rows="2"
                size=""
                success=""
                tag="input"
                type="textarea"
                validate={false}
                valueDefault=""
              />
              <p />
              <t
                className="m-1 p-1 left ml-4"
                color="primary"
                tag="button"
                type="submit"
              >
                comment
              </t>
            </form>
          </t>
        </t>
      </t>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="w-75"
          tag="div"
        >
          <t
            tag="div"
          >
            <form
              onSubmit={[Function]}
            >
              <t
                className=""
                cols="30"
                containerClass=""
                disabled={false}
                error=""
                filled={false}
                gap={false}
                group={false}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                id="reply"
                label="add comment"
                labelClass=""
                name="comment_body"
                onChange={[Function]}
                required={true}
                rows="2"
                size=""
                success=""
                tag="input"
                type="textarea"
                validate={false}
                valueDefault=""
              />
              <p />
              <t
                className="m-1 p-1 left ml-4"
                color="primary"
                tag="button"
                type="submit"
              >
                comment
              </t>
            </form>
          </t>
        </t>,
        "className": "my-3 mx-0 px-0",
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            tag="div"
          >
            <form
              onSubmit={[Function]}
            >
              <t
                className=""
                cols="30"
                containerClass=""
                disabled={false}
                error=""
                filled={false}
                gap={false}
                group={false}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                id="reply"
                label="add comment"
                labelClass=""
                name="comment_body"
                onChange={[Function]}
                required={true}
                rows="2"
                size=""
                success=""
                tag="input"
                type="textarea"
                validate={false}
                valueDefault=""
              />
              <p />
              <t
                className="m-1 p-1 left ml-4"
                color="primary"
                tag="button"
                type="submit"
              >
                comment
              </t>
            </form>
          </t>,
          "className": "w-75",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <t
                className=""
                cols="30"
                containerClass=""
                disabled={false}
                error=""
                filled={false}
                gap={false}
                group={false}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                id="reply"
                label="add comment"
                labelClass=""
                name="comment_body"
                onChange={[Function]}
                required={true}
                rows="2"
                size=""
                success=""
                tag="input"
                type="textarea"
                validate={false}
                valueDefault=""
              />
              <p />
              <t
                className="m-1 p-1 left ml-4"
                color="primary"
                tag="button"
                type="submit"
              >
                comment
              </t>
            </form>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <t
                  className=""
                  cols="30"
                  containerClass=""
                  disabled={false}
                  error=""
                  filled={false}
                  gap={false}
                  group={false}
                  icon=""
                  iconBrand={false}
                  iconClass=""
                  iconLight={false}
                  iconRegular={false}
                  id="reply"
                  label="add comment"
                  labelClass=""
                  name="comment_body"
                  onChange={[Function]}
                  required={true}
                  rows="2"
                  size=""
                  success=""
                  tag="input"
                  type="textarea"
                  validate={false}
                  valueDefault=""
                />,
                <p />,
                <t
                  className="m-1 p-1 left ml-4"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  comment
                </t>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "",
                  "cols": "30",
                  "containerClass": "",
                  "disabled": false,
                  "error": "",
                  "filled": false,
                  "gap": false,
                  "group": false,
                  "hint": undefined,
                  "icon": "",
                  "iconBrand": false,
                  "iconClass": "",
                  "iconLight": false,
                  "iconRegular": false,
                  "iconSize": undefined,
                  "id": "reply",
                  "label": "add comment",
                  "labelClass": "",
                  "name": "comment_body",
                  "onChange": [Function],
                  "required": true,
                  "rows": "2",
                  "size": "",
                  "success": "",
                  "tag": "input",
                  "type": "textarea",
                  "validate": false,
                  "valueDefault": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "comment",
                  "className": "m-1 p-1 left ml-4",
                  "color": "primary",
                  "tag": "button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "comment",
                "type": [Function],
              },
            ],
            "type": "form",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          className="my-3 mx-0 px-0"
          fluid={false}
          tag="div"
        >
          <t
            className="w-75"
            tag="div"
          >
            <t
              tag="div"
            >
              <form
                onSubmit={[Function]}
              >
                <t
                  className=""
                  cols="30"
                  containerClass=""
                  disabled={false}
                  error=""
                  filled={false}
                  gap={false}
                  group={false}
                  icon=""
                  iconBrand={false}
                  iconClass=""
                  iconLight={false}
                  iconRegular={false}
                  id="reply"
                  label="add comment"
                  labelClass=""
                  name="comment_body"
                  onChange={[Function]}
                  required={true}
                  rows="2"
                  size=""
                  success=""
                  tag="input"
                  type="textarea"
                  validate={false}
                  valueDefault=""
                />
                <p />
                <t
                  className="m-1 p-1 left ml-4"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  comment
                </t>
              </form>
            </t>
          </t>
        </t>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="w-75"
            tag="div"
          >
            <t
              tag="div"
            >
              <form
                onSubmit={[Function]}
              >
                <t
                  className=""
                  cols="30"
                  containerClass=""
                  disabled={false}
                  error=""
                  filled={false}
                  gap={false}
                  group={false}
                  icon=""
                  iconBrand={false}
                  iconClass=""
                  iconLight={false}
                  iconRegular={false}
                  id="reply"
                  label="add comment"
                  labelClass=""
                  name="comment_body"
                  onChange={[Function]}
                  required={true}
                  rows="2"
                  size=""
                  success=""
                  tag="input"
                  type="textarea"
                  validate={false}
                  valueDefault=""
                />
                <p />
                <t
                  className="m-1 p-1 left ml-4"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  comment
                </t>
              </form>
            </t>
          </t>,
          "className": "my-3 mx-0 px-0",
          "fluid": false,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <t
              tag="div"
            >
              <form
                onSubmit={[Function]}
              >
                <t
                  className=""
                  cols="30"
                  containerClass=""
                  disabled={false}
                  error=""
                  filled={false}
                  gap={false}
                  group={false}
                  icon=""
                  iconBrand={false}
                  iconClass=""
                  iconLight={false}
                  iconRegular={false}
                  id="reply"
                  label="add comment"
                  labelClass=""
                  name="comment_body"
                  onChange={[Function]}
                  required={true}
                  rows="2"
                  size=""
                  success=""
                  tag="input"
                  type="textarea"
                  validate={false}
                  valueDefault=""
                />
                <p />
                <t
                  className="m-1 p-1 left ml-4"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  comment
                </t>
              </form>
            </t>,
            "className": "w-75",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <form
                onSubmit={[Function]}
              >
                <t
                  className=""
                  cols="30"
                  containerClass=""
                  disabled={false}
                  error=""
                  filled={false}
                  gap={false}
                  group={false}
                  icon=""
                  iconBrand={false}
                  iconClass=""
                  iconLight={false}
                  iconRegular={false}
                  id="reply"
                  label="add comment"
                  labelClass=""
                  name="comment_body"
                  onChange={[Function]}
                  required={true}
                  rows="2"
                  size=""
                  success=""
                  tag="input"
                  type="textarea"
                  validate={false}
                  valueDefault=""
                />
                <p />
                <t
                  className="m-1 p-1 left ml-4"
                  color="primary"
                  tag="button"
                  type="submit"
                >
                  comment
                </t>
              </form>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <t
                    className=""
                    cols="30"
                    containerClass=""
                    disabled={false}
                    error=""
                    filled={false}
                    gap={false}
                    group={false}
                    icon=""
                    iconBrand={false}
                    iconClass=""
                    iconLight={false}
                    iconRegular={false}
                    id="reply"
                    label="add comment"
                    labelClass=""
                    name="comment_body"
                    onChange={[Function]}
                    required={true}
                    rows="2"
                    size=""
                    success=""
                    tag="input"
                    type="textarea"
                    validate={false}
                    valueDefault=""
                  />,
                  <p />,
                  <t
                    className="m-1 p-1 left ml-4"
                    color="primary"
                    tag="button"
                    type="submit"
                  >
                    comment
                  </t>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "",
                    "cols": "30",
                    "containerClass": "",
                    "disabled": false,
                    "error": "",
                    "filled": false,
                    "gap": false,
                    "group": false,
                    "hint": undefined,
                    "icon": "",
                    "iconBrand": false,
                    "iconClass": "",
                    "iconLight": false,
                    "iconRegular": false,
                    "iconSize": undefined,
                    "id": "reply",
                    "label": "add comment",
                    "labelClass": "",
                    "name": "comment_body",
                    "onChange": [Function],
                    "required": true,
                    "rows": "2",
                    "size": "",
                    "success": "",
                    "tag": "input",
                    "type": "textarea",
                    "validate": false,
                    "valueDefault": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "comment",
                    "className": "m-1 p-1 left ml-4",
                    "color": "primary",
                    "tag": "button",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "comment",
                  "type": [Function],
                },
              ],
              "type": "form",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
